app/game/Swarm/App.hs

```diff
@@ -135,9 +138,18 @@ defaultMetrics = 6543
 startMetricsThread :: Maybe Int -> Store -> IO (Either String Int)
 startMetricsThread (Just 0) _ = pure $ Left "Metrics API disabled."
 startMetricsThread mPort store = do
-  let p = fromMaybe 6543 mPort
-  _ <- WaiMetrics.forkServerWith store "localhost" p
-  pure $ Right p
+  let port = fromMaybe 6543 mPort
+  portCheck <- checkPortFree port
+  print portCheck
+  case portCheck of
+    Right () -> Right port <$ WaiMetrics.forkServerWith store "localhost" port
```

Comment:
I wanted to `catch` exception from the metrics server. But that does not work because it does `forkFinally` and then if the port binding fails in another thread, it uses `throwTo` to throw an _asynchronous exception_ to the main thread. The main thread has meanwhile exited the catch block.